Mathlib/NumberTheory/LSeries/Basic.lean

```diff
@@ -12,140 +12,216 @@ import Mathlib.Analysis.NormedSpace.FiniteDimension
 /-!
 # L-series
 
-Given an arithmetic function, we define the corresponding L-series.
+Given a sequence `f: ℕ → ℂ`, we define the corresponding L-series.
 
 ## Main Definitions
 
- * `ArithmeticFunction.LSeries` is the `LSeries` with a given arithmetic function as its
-  coefficients. This is not the analytic continuation, just the infinite series.
+ * `LSeries.term f s n` is the `n`th term of the L-series of the sequence `f` at `s : ℂ`.
+    We define it to be zero when `n = 0`.
 
- * `ArithmeticFunction.LSeriesSummable` indicates that the `LSeries`
-  converges at a given point.
+ * `LSeries f` is the L-series with a given sequence `f` as its
+    coefficients. This is not the analytic continuation (which does not necessarily exist),
+    just the sum of the infinite series if it exists and zero otherwise.
 
- * `ArithmeticFunction.LSeriesHasSum f s a` expresses that the L-series
-  of `f : ArithmeticFunction ℂ` converges (absolutely) at `s : ℂ` to `a : ℂ`.
+ * `LSeriesSummable f s` indicates that the L-series of `f` converges at `s : ℂ`.
+
+ * `LSeriesHasSum f s a` expresses that the L-series of `f` converges (absolutely)
+    at `s : ℂ` to `a : ℂ`.
 
 ## Main Results
 
- * `ArithmeticFunction.LSeriesSummable_of_le_const_mul_rpow`: the `LSeries` of an
-  arithmetic function bounded by a constant times `n^(x-1)` converges at `s` when `x < s.re`.
+ * `LSeriesSummable_of_isBigO_rpow`: the `LSeries` of a sequence `f` such that
+    `f = O(n^(x-1))` converges at `s` when `x < s.re`.
 
- * `ArithmeticFunction.zeta_LSeriesSummable_iff_one_lt_re`: the `LSeries` of `ζ`
-  (whose analytic continuation is the Riemann ζ) converges iff `1 < z.re`.
+ * `LSeriesSummable.isBigO_rpow`: if the `LSeries` of `f` is summable at `s`,
+    then `f = O(n^(re s))`.
 
 ## Tags
 
-L-series, abscissa of convergence
+L-series
 
 ## TODO
 
-* Move `zeta_LSeriesSummable_iff_one_lt_r` to a new file on L-series of specific functions
+* Move `LSeriesSummable.one_iff_one_lt_re` and `zeta_LSeriesSummable_iff_one_lt_r`
+  to a new file on L-series of specific functions
 
-* Move `LSeries_add` to a new file on algebraic operations on L-series
+* Move `LSeries_add` and friends to a new file on algebraic operations on L-series
 -/
 
 
 open scoped BigOperators
 
-namespace ArithmeticFunction
+open Complex
+
+/-!
+### The terms of an L-series
+
+We define the `n`th term evaluated at a complex number `s` of the L-series associated
+to a sequence `f : ℕ → ℂ`, `LSeries.term f s n`, and provide some basic API.
 
-open Complex Nat
+We set `LSeries.term f s 0 = 0`, and for positive `n`, `LSeries.term f s n = f n / n ^ s`.
+-/
+
+namespace LSeries
 
-/-- The L-series of an `ArithmeticFunction`. -/
```

Comment:
Below, you have to replicate a lot of stuff that says that what applies to a LSeries comes from the input function. 
In some sense, having a `LSeries` type and a map `(ℕ → ℂ) → LSeries`  (actually a `LinearMap`), and various `LSeries.summable L s`, etc. which would just take a `LSeries` L…

Reply:
Wouldn't this `LSeries` type essentially be the same as `ArithmeticFunction ℂ`? Which is what I think the consensus was to move away from. Or what would your definition of `LSeries` be?